posts/2021/06/pytorch-ignite-distributed.rst

```diff
@@ -0,0 +1,460 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here we are using the ``torch.distributed.launch`` script in order to
+spawn the processes:
+
+.. code:: bash
+
+   python -m torch.distributed.launch --nproc_per_node 2 --use_env ignite_idist.py --backend gloo
+
+With horovodrun
+~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+   horovodrun -np 4 -H hostname1:2,hostname2:2 python ignite_idist.py --backend horovod
+
+.. note::
+   
+   In order to run this example and to avoid the installation procedure, you can pull one of PyTorch-Ignite's `docker image with pre-installed Horovod <https://github.com/pytorch/ignite/blob/master/docker/hvd/Dockerfile.hvd-base>`__. It will include Horovod with ``gloo`` controller and ``nccl`` support.
+
+   .. code:: bash
+
+      docker run --gpus all -it -v $PWD:/workspace/project --network=host --shm-size 16G pytorchignite/hvd-vision:latest /bin/bash
+      cd project
+
+With slurm
+~~~~~~~~~~~~
+
+The same result can be achieved by using ``slurm`` without any
+modification to the code:
+
+.. code:: bash
+
+   srun --nodes=2
+        --ntasks-per-node=2 
+        --job-name=pytorch-ignite 
+        --time=00:01:00  
+        --partition=gpgpu 
+        --gres=gpu:2
+        --mem=10G 
+        python ignite_idist.py --backend nccl
+
+or using ``sbatch script.bash`` with the script file ``script.bash``:
+
+.. code:: shell
+
+   #!/bin/bash
+   #SBATCH --job-name=pytorch-ignite
+   #SBATCH --output=slurm_%j.out
+   #SBATCH --nodes=2
+   #SBATCH --ntasks-per-node=2
+   #SBATCH --time=00:01:00
+   #SBATCH --partition=gpgpu
+   #SBATCH --gres=gpu:2
+   #SBATCH --mem=10G
+
+   srun python ignite_idist.py --backend nccl
+
+Closing Remarks
+===============
+
+As we saw through the above examples, managing multiple configurations
+and specifications for distributed computing has never been easier. In
+just a few lines we can parallelize and execute code wherever it is
+while maintaining control and simplicity.
+
+References
+----------
+
+-  `idist-snippets <https://github.com/pytorch-ignite/idist-snippets/>`__:
+   complete code used in this post.
+
+-  `why-ignite <https://github.com/sdesrozis/why-ignite>`__: examples
+   with distributed data parallel: native pytorch, pytorch-ignite,
+   slurm.
+
+-  `CIFAR10
+   example <https://github.com/pytorch/ignite/tree/master/examples/contrib/cifar10>`__
+   of distributed training on CIFAR10 with muliple configurations: 1 or
+   multiple GPUs, multiple nodes and GPUs, TPUs.
+
+Next Steps
+----------
+
+-  If you want to learn more about PyTorch-Ignite or have any further
+   queries, here is our `GitHub <https://github.com/pytorch/ignite>`__,
+   `documentation <https://pytorch.org/ignite/>`__ and
+   `Discord <https://discord.com/invite/djZtm3EmKj>`__.
+
+-  PyTorch-Ignite is currently maintained by a team of volunteers and we
+   are looking for more contibutors.
+   `Here <https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md>`__
+   is how you can contribute.
```

Comment:
The "Here" being a link is a little easy to miss (our color contrast on links isn't great). Maybe rewrite this as: 
```
See `CONTRIBUTE.md <https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md>`__
for how you can contribute.
```